app/views/contexts/section_summary_context.py

```diff
@@ -113,17 +116,34 @@ def _title_for_location(self, location):
         )
         return title
 
-    def _custom_summary_elements(self, section_summary, current_location, section):
+    def _custom_summary_elements(
+        self, section_summary, current_location, section, routing_path
+    ):
         for summary_element in section_summary:
             if summary_element["type"] == "List":
                 yield self._list_summary_element(
-                    summary_element, current_location, section
+                    summary_element, current_location, section, routing_path
                 )
 
-    def _list_summary_element(self, summary, current_location, section) -> Mapping:
+    def _list_summary_element(
+        self, summary, current_location, section, routing_path
+    ) -> Mapping:
         current_list = self._list_store[summary["for_list"]]
-        list_collector_block = self._schema.get_list_collector_for_list(
-            section, for_list=summary["for_list"]
+
+        list_collector_blocks_on_path = [
+            list_collector_block
+            for list_collector_block in self._schema.get_list_collectors_for_list(
+                section, for_list=summary["for_list"]
+            )
+            if list_collector_block["id"] in routing_path.block_ids
+        ]
+
+        list_collector_block = (
+            list_collector_blocks_on_path[0]
+            if list_collector_blocks_on_path
+            else self._schema.get_list_collector_for_list(
+                section, for_list=summary["for_list"]
+            )
         )
```

Comment:
This can actually be:
```py
        list_collector_block = (
            list_collector_blocks_on_path[0]
            if list_collector_blocks_on_path
            else None
        )
```
This will allow you to remove the need of routing path from `self._add_link`.

Reply:
Done, thanks!